From: Tetsu Yamamoto <tyamamot@redhat.com>
Date: Mon, 4 Aug 2008 15:48:41 -0400
Subject: [xen] ia64: don't warn for EOI-ing edge triggered intr
Message-id: 20080804194841.8452.10651.sendpatchset@pq0-1.lab.bos.redhat.com
O-Subject: [RHEL5.3 PATCH 9/10] xen-ia64: Don't warn for EOI-ing an edge triggered interrupt
Bugzilla: 430219
RH-Acked-by: Bill Burns <bburns@redhat.com>

bz430219
# HG changeset patch
# User Alex Williamson <alex.williamson@hp.com>
# Date 1192638318 21600
# Node ID ab89a931f94f6ddea0aba876ff99d6c384d40d92
# Parent  2863852e02f6d5202f62e792f1768941e81f2cbd
[IA64] Don't warn for EOI-ing an edge triggered interrupt.

Edge triggered interrupts don't need an EOI but EOI-ing them is harmless
according to the IOSAPIC documentation.  Do not emit a warning in such case.

Signed-off-by: Tristan Gingold <tgingold@free.fr>

diff --git a/arch/ia64/vmx/viosapic.c b/arch/ia64/vmx/viosapic.c
index bed4e2b..f951aa0 100644
--- a/arch/ia64/vmx/viosapic.c
+++ b/arch/ia64/vmx/viosapic.c
@@ -126,8 +126,9 @@ static void viosapic_update_EOI(struct viosapic *viosapic, int vector)
 
     if ( !test_and_clear_bit(redir_num, &viosapic->isr) )
     {
-        gdprintk(XENLOG_WARNING, "redir %d not set for %d EOI\n",
-                 redir_num, vector);
+        if ( viosapic->redirtbl[redir_num].trig_mode == SAPIC_LEVEL )
+            gdprintk(XENLOG_WARNING, "redir %d not set for %d EOI\n",
+                     redir_num, vector);
         return;
     }
     service_iosapic(viosapic);
